tools/junit-openfeature/src/main/java/dev/openfeature/contrib/tools/junitopenfeature/TestProvider.java

```diff
@@ -0,0 +1,128 @@
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    public TestProvider(ExtensionContext.Namespace namespace, Map<String, Flag<?>> flags) {
+        flagsMap.put(namespace, flags);
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        super.initialize(evaluationContext);
+        state = ProviderState.READY;
+        log.debug("finished initializing provider, state: {}", state);
+    }
+
+    /**
+     * Updating provider flags configuration, replacing existing flags.
+     * @param flags the flags to use instead of the previous flags.
+     */
+    public void addFlags(ExtensionContext.Namespace namespace, Map<String, Flag<?>> flags) {
+        flagsMap.put(namespace, flags);
+    }
+
+    @Override
+    public ProviderEvaluation<Boolean> getBooleanEvaluation(String key, Boolean defaultValue,
+                                                            EvaluationContext evaluationContext) {
+        return getEvaluation(key, evaluationContext, Boolean.class);
+    }
+
+    @Override
+    public ProviderEvaluation<String> getStringEvaluation(String key, String defaultValue,
+                                                          EvaluationContext evaluationContext) {
+        return getEvaluation(key, evaluationContext, String.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Integer> getIntegerEvaluation(String key, Integer defaultValue,
+                                                            EvaluationContext evaluationContext) {
+        return getEvaluation(key, evaluationContext, Integer.class);
+    }
+
+    @Override
+    public ProviderEvaluation<Double> getDoubleEvaluation(String key, Double defaultValue,
+                                                          EvaluationContext evaluationContext) {
+        return getEvaluation(key, evaluationContext, Double.class);
+    }
+
+    @SneakyThrows
+    @Override
+    public ProviderEvaluation<Value> getObjectEvaluation(String key, Value defaultValue,
+                                                         EvaluationContext evaluationContext) {
+        return getEvaluation(key, evaluationContext, Value.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    private <T> ProviderEvaluation<T> getEvaluation(
+            String key, EvaluationContext evaluationContext, Class<?> expectedType
+    ) throws OpenFeatureError {
+        if (!ProviderState.READY.equals(state)) {
+            if (ProviderState.NOT_READY.equals(state)) {
+                throw new ProviderNotReadyError("provider not yet initialized");
+            }
+            throw new GeneralError("unknown error");
+        }
+        Map<ExtensionContext.Namespace, Map<String, Flag<?>>> flagsMap1 = flagsMap;
```

Comment:
nit - do we need this variable ?

Reply:
ah f... i forgot to remove my debugging variables, i will inline them